test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c

```diff
@@ -593,6 +699,50 @@ void test_DNS_TreatNBNS_success_empty_char_nbns_name( void )
                    300,
                    1234 );
 }
+
+/**
+ * @brief ensures that when pucPayload + dnsNBNS_ENCODED_NAME_LENGTH = 41U = 'A'
+ *        we break out of the loop and nothing is sent over the network if the
+ *        whole name if not in capital letters and terminated with '\0'
+ */
+void test_DNS_TreatNBNS_success_empty_char_nbns_name2( void )
```

Comment:
naming?

Reply:
same, 1 2 3 is to hit the same conditional branch in different ways, the functionality is the same, thus the name